src/neo/Network/P2P/TaskManager.cs

```diff
@@ -346,8 +349,13 @@ internal protected override bool ShallDrop(object message, IEnumerable queue)
         {
             if (!(message is TaskManager.NewTasks tasks)) return false;
             // Remove duplicate tasks
-            if (queue.OfType<TaskManager.NewTasks>().Any(x => x.Payload.Type == tasks.Payload.Type && x.Payload.Hashes.SequenceEqual(tasks.Payload.Hashes))) return true;
-            return false;
+            return !taskHashes.TryAdd(tasks.Payload.PayLoadHash, tasks.Payload.Type);
```

Comment:
In order to reduce collisions, I think that it's better one list per PayloadType.